cli/src/pcluster/validators/ec2_validators.py

```diff
@@ -331,9 +335,22 @@ def _validate(
         is_flexible: bool,
         subnet: str,
         capacity_type: CapacityType,
+        os,
     ):
         if capacity_reservation_id:
             capacity_reservation = AWSApi.instance().ec2.describe_capacity_reservations([capacity_reservation_id])[0]
+            cr_platform = capacity_reservation.instance_platform()
+            if CAPACITY_RESERVATION_OS_MAP.get(cr_platform):
```

Comment:
the way the validator fails, may be misleading. Example:
1. user wants to create cluster with OS AL23
2. user creates a reservation with platform WHATEVER_BUT_NOT_THE_CORRECT_ONE
3. the error message says to create a reservation with one of Linux/Unix, Ubuntu Pro, RHEL
4. it creates a RHEL reservation
5. now the validator fails with the helpful message, saying that AL23 requires Linux/Unix platform.

Proposed solution: fail with just one error message saying that the capacity reservation {CR_ID} has platform {PLATFORM}, which is not compatible with the cluster OS {OS}. Please use a reservation with platform {CORRECT_PLATFORM}

What do you think?

Reply:
will make changes